src/Interpreters/WindowDescription.cpp

```diff
@@ -160,7 +160,7 @@ void WindowFrame::checkValid() const
         bool begin_less_equal_end;
         if (begin_preceding && end_preceding)
         {
-            begin_less_equal_end = begin_offset >= end_offset;
+            begin_less_equal_end = begin_offset.get<Int64>() >= end_offset.get<Int64>();
```

Comment:
Oh, so the comparison operator of Field was not an arithmetic comparison but something for sorting.... maybe we should remove it to avoid accidental errors of this kind.

Your fix looks dubious because offsets can be float, and `get<>` (again, a bad non-descriptive name) is a reinterpret_cast  with type check in debug. Looks like `FieldVisitorAccurateLess` should be used here.

Reply:
Even if both boundaries have `BoundaryType::Offset` ?
Ok, changed to visitor